prometheus_aws_guardduty_exporter/collector.py

```diff
@@ -23,30 +43,28 @@ def collect(self):
             labels=["region"])
 
         # Customize the boto client config
-        botoConfig = botocore.client.Config(connect_timeout=2, read_timeout=10, retries={"max_attempts": 2})
+        self.botoConfig = botocore.client.Config(connect_timeout=2, read_timeout=10, retries={"max_attempts": 2})
 
+        threads = []
         for region in self.regions:
-            try:
-                # Scrape GuardDuty statistics from the region
-                regionStats = self._collectMetricsByRegion(region, botoConfig)
+            # We start one thread per region.
+            process = threading.Thread(target=self.scrape_metric_in_region, args=[currentFindingsMetric, region])
```

Comment:
I'm not a big fan of this design because it introduce a lock to modify the metric in several different threads. I believe an easier and cleaner approach would be using a [thread pool](https://docs.python.org/2/library/multiprocessing.html#module-multiprocessing.pool):

1. Create a pool of X threads
2. Call `pool.apply_async(self. _collectMetricsByRegion, [region])`: it returns an [`AsyncResult`](https://docs.python.org/2/library/multiprocessing.html#multiprocessing.pool.AsyncResult) which you will store in a `results` array
3. Iterate over the `results` array to get the result (`result.get()`)

If I'm not missing antying, in this scenario `scrape_metric_in_region()` should not be required at all.